paymentsheet/src/main/java/com/stripe/android/paymentsheet/repositories/CustomerAdapter.kt

```diff
@@ -0,0 +1,150 @@
+package com.stripe.android.paymentsheet.repositories
+
+import android.content.Context
+import androidx.annotation.RestrictTo
+import com.stripe.android.ExperimentalSavedPaymentMethodsApi
+import com.stripe.android.model.PaymentMethod
+
+/**
+ * [CustomerAdapter] A "bridge" from wallet mode to your backend to fetch Customer-related
+ * information. Typically, you will not need to implement this interface yourself. You should
+ * instead use [CustomerAdaper.create], which manages retrieving and updating a Stripe customer for
+ * you.
+ *
+ * Implement this interface if you would prefer retrieving and updating your Stripe customer object
+ * via your own backend instead of using the default implementation.
+ *
+ * Use [CustomerAdapter.create] to create an instance of a [CustomerAdapter].
+ */
+@ExperimentalSavedPaymentMethodsApi
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+interface CustomerAdapter {
+    /**
+     * Retrieves a list of payment methods attached to the customer provided by the
+     * [CustomerEphemeralKeyProvider]
+     */
+    suspend fun retrievePaymentMethods(): Result<List<PaymentMethod>>
+
+    /**
+     * Attaches a payment method to a customer
+     * @param paymentMethodId, the payment method to attach to the customer provided by the
+     * [CustomerEphemeralKeyProvider]
+     */
+    suspend fun attachPaymentMethod(paymentMethodId: String)
+
+    /**
+     * Detaches the given payment method from the customer provided by the
+     * [CustomerEphemeralKeyProvider]
+     * @param paymentMethodId, the payment method to detach from the customer
+     */
+    suspend fun detachPaymentMethod(paymentMethodId: String)
+
+    /**
+     * Set the selected payment method option.
+     * @param paymentOption, the [PersistablePaymentMethodOption] to save to the data store. If
+     * null, the selected payment method option will be cleared from the data store.
+     */
+    suspend fun setSelectedPaymentMethodOption(paymentOption: PersistablePaymentMethodOption?)
+
+    /**
+     * Fetch the persisted payment method option from the data store. If null, the customer does
+     * not have a default saved payment method.
+     */
+    suspend fun fetchSelectedPaymentMethodOption(): Result<PersistablePaymentMethodOption?>
+
+    /**
+     * Returns a client secret configured to attach a new payment method to a customer.
+     * This will call your backend to retrieve a client secret if you have provided a
+     * setupIntentClientSecretProvider in the init call.
+     */
+    suspend fun setupIntentClientSecretForCustomerAttach(): Result<String>
+
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
```

Comment:
I _think_ the rule is, anytime we add restrict to, we should also add the experimental annotation.